site/app/libraries/database/DatabaseQueries.php

```diff
@@ -3809,4 +3809,18 @@ public function createEmail($email_data, $recipient){
               created)
             VALUES(?, ?, ?, NOW())", $parameters);
     }
+
+    /**
+     * Gives true if thread is locked
+     * @param $thread_id
+     * @return bool
+     */
+    public function isThreadLocked($thread_id){
+
+        $this->course_db->query('SELECT lock_thread_date FROM threads WHERE id = ?', [$thread_id]);
+        if(empty($this->course_db->rows()[0]['lock_thread_date'])){
+            return false;
+        }
+        return $this->course_db->rows()[0]['lock_thread_date'] < date("Y-m-d H:i:S");
```

Comment:
Not sure about it but probably there are better ways to compare dates, like `strtotime` or `date_diff`.
You may want to reuse `calculateDayDiff` 
https://github.com/Submitty/Submitty/blob/98ddf16531c51e9b0c97b8949b15edca0e2d82f9/site/app/libraries/DateUtils.php#L26

Reply:
PHP can itself compare two timestamp so It will make no sense to use calculateDayDiff

Reply:
@drumilpatel2000 Ohhh true. I misunderstood the function. Sorry for being dense.